### DISCUSSION TITLE
ListView cannot focus

### DISCUSSION THREAD

**Discussion by Decodetalkers** [slint-ui/slint#1718] [Q&A] [answered]
I want to focus on the item and select the item in the listview, and listen to the keyboard event of enter, but I cannot find how to use up and down to select the item in listview

**Comment by ogoffart** [comment, marked as answer]
The StandardListView widget actually handle the up and down keys.  Here is the implementation in case you want to do the same in your custom ListView:

https://github.com/slint-ui/slint/blob/b1b4a443a47b16aeae5cac0bf06a05da7a932a21/internal/compiler/widgets/fluent/std-widgets.slint#L496-L528

Edit: The part that handle the keys is the FocusScope part, line 516

**Reply by Decodetalkers** [reply]
> The StandardListView widget actually handle the up and down keys. Here is the implementation in case you want to do the same in your custom ListView:
> 
> https://github.com/slint-ui/slint/blob/b1b4a443a47b16aeae5cac0bf06a05da7a932a21/internal/compiler/widgets/fluent/std-widgets.slint#L496-L528
> 
> Edit: The part that handle the keys is the FocusScope part, line 516

Another question, I use the textedit and a listview, after I edit in textedit, then focus is in textedit, how to change the focus to listview again? or set the focus back to the root

I see, focus()
